packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesModule.java

```diff
@@ -129,7 +131,11 @@ public void checkForUpdateAsync(final Promise promise) {
         return;
       }
 
-      FileDownloader.downloadManifest(updatesService.getConfiguration(), getContext(), new FileDownloader.ManifestDownloadCallback() {
+      DatabaseHolder databaseHolder = updatesService.getDatabaseHolder();
+      JSONObject extraHeaders = ManifestMetadata.getServerDefinedHeaders(databaseHolder.getDatabase(), updatesService.getConfiguration());
+      databaseHolder.releaseDatabase();
```

Comment:
Yeah, I think that makes sense and is worth doing in a separate PR. In some instances the "get" and "release" calls are spread across different contexts because of callbacks and such, so it's hard to have something like that work 100% of the time, but wouldn't hurt to add another layer of safety.